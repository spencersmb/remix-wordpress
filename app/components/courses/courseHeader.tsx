import Imgix, { Picture, Source } from 'react-imgix'
import { LazyLoadImage } from 'react-lazy-load-image-component'
import { staticImages } from '~/lib/imgix/data'
import LazyImageBase from '../images/lazyImage-base'
import HeyTeela from '../svgs/heyTeela'

interface Props {

}


function CourseHeader(props: Props) {

  return (
    <div className='grid grid-flow-row grid-rows-[auto_auto_1fr_1fr_1fr] bg-neutral-50 grid-cols-mobile gap-x-5 tablet:grid-cols-tablet tablet:gap-x-5 laptop:grid-rows-[minmax(60px,auto)_1fr_1fr_minmax(60px,auto)_minmax(60px,auto)] desktop:grid-cols-desktop'>

      {/* INTRO TEXT */}
      <div className='relative z-30 flex flex-col col-span-2 col-start-2 row-span-1 row-start-1 pb-8 tablet:col-start-2 tablet:col-span-7 tablet:pr-8 laptop:col-start-3 laptop:col-span-6 laptop:row-span-4 laptop:row-end-4 desktop:col-span-5 desktop:col-start-2 desktop:pr-0 desktop:pl-8 desktopXl:px-0 desktopXl:col-span-4 desktopXl:col-start-3'>
        <div className='flex-1 max-w-[60%] pt-8 '>
          <LazyLoadImage
            className='w-full'
            key={'Online Course Scribbles'}
            alt={'Every Tuesday Online Courses'}
            effect="blur"
            placeholderSrc={headerData.scribble.placeholder}
            src={headerData.scribble.src}
          />
        </div>
        <h1 className='text-7xl font-sentinel__SemiBoldItal text-sage-700 laptop:max-w-[470px] desktop:text-9xl desktop:mb-8'>
          Online Courses
        </h1>
        <p className='pt-4 text-lg text-sage-600'>
          {headerData.content}
        </p>
      </div>

      {/* BACKGROUND */}
      <div className='col-end-[-1] col-start-3 row-span-3 row-start-2 bg-sage-200 tablet:row-start-1 tablet:col-start-10 laptop:row-start-1 laptop:row-end-4 desktop:col-start-8' />

      {/* PROFILE IMAGE */}
      <div className='relative flex justify-center row-span-4 row-start-2 my-10 col-span-full tablet:row-start-1 tablet:col-start-8 tablet:col-span-6 laptop:row-start-2 laptop:row-span-4 laptop:m-0 desktopXl:col-span-5 desktopXl:col-start-8'>

        <div className='relative bg-white rotate-[8deg] w-full max-w-[202px] m-auto border-[10px] border-white shadow-md z-10 laptop:max-w-[300px] desktop:max-w-[390px] desktop:border-[20px]'>

          {/* TEELA NAME + ARROW */}
          <div className='absolute top-[13%] left-[-100px] w-[175px] z-20 tablet:left-[-120px] laptop:w-[250px] desktop:w-[300px] laptop:top-[5%]'>
            <HeyTeela />
          </div>

          {/* BLACK PIN */}
          <div className='absolute max-w-[76px] z-20 top-[-18px] left-[45%] translate-x-[-50%] desktop:max-w-[100px]'>
            <LazyLoadImage
              key={'blackPin'}
              alt={'Every Tuesday Hand Made Black Pin'}
              effect="blur"
              placeholderSrc={staticImages.assets.pins.black_1.placeholder}
              src={staticImages.assets.pins.black_1.src}
            />
          </div>

          <div className='relative z-10 '>
            <LazyImageBase
              id={'teela'}
              alt={'Every Tuesday: Hey I\'m Teela, your course instructor.'}
              key={'Online Course Author'}
              image={
                {
                  width: '800',
                  height: '1367',
                  altTitle: 'Every Tuesday: Hey I\'m Teela, your course instructor.',
                  sourceUrl: headerData.profileImage.src,
                  placeholder: headerData.profileImage.placeholder,
                  srcSet: '',
                  sizes: 'string'
                }
              }
            />
          </div>
        </div>

        {/* GREEN TEXTURE */}
        <div className='absolute top-[-30px] left-auto right-[-30px] w-[300px] z-0 rotate-[-175deg] tablet:w-[500px] tablet:right-[-220px] tablet:rotate-[45deg] laptop:top-[-50px] laptop:right-[-90px] laptop:w-[400px] desktop:w-[545px]'>
          <Picture >
            <Source
              attributeConfig={{
                src: 'data-src',
                srcSet: 'data-srcset',
                sizes: 'data-sizes'
              }}
              src={staticImages.textures.greenLarge.src}
              width={600}
              htmlAttributes={{ media: "(min-width: 320px)" }}
            />
            <Imgix
              className="lazyload"
              src={staticImages.textures.greenLarge.src}
              attributeConfig={{
                src: 'data-src',
                srcSet: 'data-srcset',
                sizes: 'data-sizes'
              }}
              imgixParams={{ w: 100 }}
              htmlAttributes={{
                src: staticImages.textures.greenLarge.placeholder, // low quality image here
              }} />
          </Picture>
        </div>
      </div>
    </div>
  )
}

export default CourseHeader

const headerData = {
  title: 'Online Courses',
  content: 'I’ve taught over 200,000 students and my design + lettering videos on YouTube have accumulated over 19 million views. I love sharing what I’ve learned over my career and motivating others to create something new every week.',
  profileImage: {
    src: staticImages.profiles.teela.vertical.src,
    placeholder: staticImages.profiles.teela.vertical.placeholder,
  },
  scribble: {
    src: staticImages.scribbles.scribble_4.src,
    placeholder: staticImages.scribbles.scribble_4.src,
  },
  textureImage: {
    ...staticImages.textures.greenLarge,
  }
}