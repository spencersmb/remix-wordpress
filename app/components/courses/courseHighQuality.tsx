import React from 'react'
import BulletListSvg from '../svgs/bulletList'
import PictureFrameSvg from '../svgs/pictureFrameSvg'
import VideoCameraSvg from '../svgs/videoCamera'

interface Props { }
function getBgColors(position: number): string {
  switch (position) {
    case 1:
      return 'bg-secondary-200'

    case 2:
      return 'bg-primary-200'

    default:
      return 'bg-sage-200'
  }
}
function CourseHighQuality(props: Props) {

  return (
    <div className='grid grid-flow-row py-8 bg-sage-50 grid-cols-mobile gap-x-5 tablet:py-24 tablet:grid-cols-tablet tablet:gap-x-5 desktop:grid-cols-desktop'>

      <div className='col-span-2 col-start-2 mb-8 text-center text-sage-700 tablet:col-start-4 tablet:col-span-8 laptop:col-start-5 laptop:col-span-6 laptop:mb-16'>
        <h3 className='mb-8 text-5xl font-sentinel__SemiBoldItal desktop:text-6xl desktop:max-w-[500px] desktop:mx-auto'>High quality video & class assets</h3>
        <p className='tablet:text-lg laptop:text-xl desktop:text-2xl'>We want to provide the best environment for you to learn with high quality videos that show every step with every tool Teela uses during the course so you can follow along perfectly.</p>
      </div>


      <div className='grid grid-cols-1 col-span-2 col-start-2 tablet:grid-cols-3 tablet:col-start-2 tablet:col-span-12 laptop:px-[2%] desktop:max-w-[1000px] desktop:mx-auto'>
        {qualityList.map((item, index) => {

          let bgColor = getBgColors(index)

          return (
            <div key={index} className="flex items-center p-4 mb-4 bg-white rounded-lg shadow-xs tablet:mx-4">
              <div className={`${bgColor} rounded-lg p-2 w-[53px] laptop:p-4 laptop:w-[73px]`}>
                {item.svg}
              </div>
              <div className='flex-1 pl-4 font-bold leading-6 laptop:text-lg text-sage-600'>
                <h4>{item.title}</h4>
              </div>
            </div>
          )
        })}
      </div>

    </div>
  )
}

export default CourseHighQuality

const qualityList = [
  {
    title: 'Pre-recorded videos',
    svg: <VideoCameraSvg stroke={`#000000`} />
  },
  {
    title: 'Step-by-step instructions',
    svg: <BulletListSvg stroke={`#000000`} />
  },
  {
    title: 'Project Based Tutorials',
    svg: <PictureFrameSvg stroke={`#000000`} />
  }
]