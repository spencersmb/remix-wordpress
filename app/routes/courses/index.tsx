const Courses = () => {
  return (
    <div>
      Courses Page
    </div>
  )
}

export default Courses
