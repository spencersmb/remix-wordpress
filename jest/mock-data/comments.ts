import { mockAuthorData } from "./posts"

export const mockComment: IPostComment = {
  databaseId: 218427,
  // approved: true,
  parent: null,
  id: "Y29tbWVudDoyMTg0Mjc=",
  author: {
    id: "YWRtaW5pc3RyYXRvcjox",
    databaseId: "1",
    name: "Spencer Bigum",
    gravatar: {
      url: "https://secure.gravatar.com/avatar/9119dc766dbfe9f6b05b7f39f9dbd9ce?s=96&d=mm&r=g"
    }
  },
  date: "2022-01-27 21:08:02",
  content: "<p>Comment Body</p>\n",
  replies: []
}

export const mockComments: IComments = {
    pageInfo: {
    endCursor: "YXJyYXljb25uZWN0aW9uOjIxODQyNA==",
    hasNextPage: true
  },
  list: [
    {
      databaseId: 218427,
      // approved: true,
      parent: null,
      id: "Y29tbWVudDoyMTg0Mjc=",
      author: {
        id: "YWRtaW5pc3RyYXRvcjox",
        databaseId: "1",
        name: "Spencer Bigum",
        gravatar: {
          url: "https://secure.gravatar.com/avatar/9119dc766dbfe9f6b05b7f39f9dbd9ce?s=96&d=mm&r=g"
        }
      },
      date: "2022-01-27 21:08:02",
      // commentedOn: {
      //   node: {
      //     id: "cG9zdDo1NDM0",
      //     status: "publish"
      //   }
      // },
      content: "<p>4</p>\n",
      replies: []
      },
    {
      databaseId: 218426,
      // approved: true,
      parent: null,
      id: "Y29tbWVudDoyMTg0MjY=",
      author: {
        databaseId: "1",
        id: "YWRtaW5pc3RyYXRvcjox",
        name: "Spencer Bigum",
        gravatar: {
          url: "https://secure.gravatar.com/avatar/9119dc766dbfe9f6b05b7f39f9dbd9ce?s=96&d=mm&r=g"
        }
      },
      date: "2022-01-27 21:07:22",
      // commentedOn: {
      //   node: {
      //     id: "cG9zdDo1NDM0",
      //     status: "publish"
      //   }
      // },
      content: "<p>3</p>\n",
      replies: [
          {
            id: "Y29tbWVudDoyMTg0MzM=",
            databaseId: 218433,
            content: "<p>two test</p>\n",
            date: "2022-01-28 01:31:37",
            parent: 218426,
            author: {
              databaseId: "2",
              id: "YWRtaW5pc3RyYXRvcjoxMg==",
              name: "twotest",
              gravatar: {
                url: "https://secure.gravatar.com/avatar/9119dc766dbfe9f6b05b7f39f9dbd9ce?s=96&d=mm&r=g"
              }
            }
          },
          {
            id: "Y29tbWVudDoyMTg0MzI=",
            databaseId: 218432,
            content: "<p>test</p>\n",
            date: "2022-01-28 01:26:12",
            parent: 218426,
            author: {
              databaseId: "1",
              id: "YWRtaW5pc3RyYXRvcjox",
              name: "Spencer M Bigum",
              gravatar: {
              url: "https://secure.gravatar.com/avatar/9119dc766dbfe9f6b05b7f39f9dbd9ce?s=96&d=mm&r=g"
              }
            }
            },
          {
            id: "Y29tbWVudDoyMTg0Mjg=",
            databaseId: 218428,
            content: "<p>tes</p>\n",
            date: "2022-01-27 21:08:26",
            parent: 218426,
            author: {
              databaseId: "3",
              id: "YWRtaW5pc3RyYXRvcjoxMw==",
              name: "Teela Bigum",
              gravatar: {
                url: "https://secure.gravatar.com/avatar/aac786a8b226660a1064233ededdda6b?s=96&d=mm&r=g"
              }
            }
          }
        ]
      },
    {
      databaseId: 218424,
      // approved: true,
      parent: null,
      id: "Y29tbWVudDoyMTg0MjQ=",
      author: {
        databaseId: "1",
        id: "YWRtaW5pc3RyYXRvcjox",
        name: "Spencer",
        gravatar: {
          url: "https://secure.gravatar.com/avatar/9119dc766dbfe9f6b05b7f39f9dbd9ce?s=96&d=mm&r=g"
        }
      },
      date: "2022-01-27 21:05:42",
      // commentedOn: {
      //   node: {
      //     id: "cG9zdDo1NDM0",
      //     status: "publish"
      //   }
      // },
      content: "<p>test</p>\n",
      replies: []
      }
  ]
}

export const mockPostCommentRaw: IPostCommentRaw = {
  author: {
    node: {
      ...mockAuthorData,
      databaseId: '1',
      id: '1',
      gravatar: {
        url: 'https://www.gravatar.com/avatar/00000000000000000000000000000000?d=mm&s=24',
      }
    }
  },
  content: "This is a comment",
  databaseId: 1,
  approved: false,
  parent: null,
  commentedOn:{
    node:{
      databaseId: 1
    }
  },
  date: "2018-04-23T15:00:00",
  id: '1',
  replies: {
    edges:[
      {
        node: {
          author: {node: {
            ...mockAuthorData,
            databaseId: '1',
            id: '1',
            gravatar: {
              url: 'https://www.gravatar.com/avatar/00000000000000000000000000000000?d=mm&s=24',
            }
          }},
          content: 'comment reply',
          databaseId: 10,
          date: '2018-04-23T15:00:00',
          id: '10',
          parent:{
            node: {
              databaseId: 1
            }
          }
        }
      }
    ]
  }
}