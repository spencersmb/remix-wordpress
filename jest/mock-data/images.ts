 export const mockFeaturedImage: IFeaturedImage = {
    altText: '',
    sourceUrl: '',
    mediaDetails: {
      sizes: [],
      height: 0,
      width: 0,
    },
    id: '',
    mimeType: '',
    sizes: '',
    srcSet: '',
  }
  export const mockFeatureImageComplete: IFeaturedImage = {
      mediaDetails: {
        height: 0,
        width:  0,
        sizes: [
          {
            width: "1000",
            file: "create-candy-cane-lettering-in-procreate-hero-1000x749.jpg",
            height: "749",
            name: "headless_post_thumbnail",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-1000x749.jpg",
            mimeType: "image/jpeg"
          },
          {
            width: "600",
            file: "create-candy-cane-lettering-in-procreate-hero-600x360.jpg",
            height: "360",
            name: "headless_resource_image",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-600x360.jpg",
            mimeType: "image/jpeg"
          },
          {
            width: "20",
            file: "create-candy-cane-lettering-in-procreate-hero-20x20.jpg",
            height: "20",
            name: "placeholder",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-20x20.jpg",
            mimeType: "image/jpeg"
          },
          {
            width: "1000",
            file: "create-candy-cane-lettering-in-procreate-hero-1000x749.jpg",
            height: "749",
            name: "headless_ipad",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-1000x749.jpg",
            mimeType: "image/jpeg"
          },
          {
            width: "1440",
            file: "create-candy-cane-lettering-in-procreate-hero-1440x810.jpg",
            height: "810",
            name: "headless_post_feature_image",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-1440x810.jpg",
            mimeType: "image/jpeg"
          },
          {
            width: "150",
            file: "create-candy-cane-lettering-in-procreate-hero-150x150.jpg",
            height: "150",
            name: "thumbnail",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-150x150.jpg",
            mimeType: "image/jpeg"
          },
          {
            width: "500",
            file: "create-candy-cane-lettering-in-procreate-hero-500x281.jpg",
            height: "281",
            name: "medium",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-500x281.jpg",
            mimeType: "image/jpeg"
          },
          {
            width: "1024",
            file: "create-candy-cane-lettering-in-procreate-hero-1024x576.jpg",
            height: "576",
            name: "large",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-1024x576.jpg",
            mimeType: "image/jpeg"
          },
          {
            width: "1536",
            file: "create-candy-cane-lettering-in-procreate-hero-1536x864.jpg",
            height: "864",
            name: "1536x1536",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-1536x864.jpg",
            mimeType: "image/jpeg"
          },
          {
            width: "100",
            file: "create-candy-cane-lettering-in-procreate-hero-100x56.jpg",
            height: "56",
            name: "et_thumbnail",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-100x56.jpg",
            mimeType: "image/jpeg"
          },
          {
            width: "550",
            file: "create-candy-cane-lettering-in-procreate-hero-550x550.jpg",
            height: "550",
            name: "readanddigest_square",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-550x550.jpg",
            mimeType: "image/jpeg"
          },
          {
            width: "800",
            file: "create-candy-cane-lettering-in-procreate-hero-800x600.jpg",
            height: "600",
            name: "readanddigest_landscape",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-800x600.jpg",
            mimeType: "image/jpeg"
          },
          {
            width: "600",
            file: "create-candy-cane-lettering-in-procreate-hero-600x800.jpg",
            height: "800",
            name: "readanddigest_portrait",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-600x800.jpg",
            mimeType: "image/jpeg"
          },
          {
            width: "1200",
            file: "create-candy-cane-lettering-in-procreate-hero-1200x675.jpg",
            height: "675",
            name: "readanddigest_post_feature_image",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-1200x675.jpg",
            mimeType: "image/jpeg"
          },
          {
            width: "128",
            file: "create-candy-cane-lettering-in-procreate-hero-128x86.jpg",
            height: "86",
            name: "readanddigest_thumb",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-128x86.jpg",
            mimeType: "image/jpeg"
          },
          {
            width: "1200",
            file: "create-candy-cane-lettering-in-procreate-hero-1200x580.jpg",
            height: "580",
            name: "readanddigest_single_post_title",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-1200x580.jpg",
            mimeType: "image/jpeg"
          },
          {
            width: "1100",
            file: "create-candy-cane-lettering-in-procreate-hero-1100x550.jpg",
            height: "550",
            name: "readanddigest_large_width",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-1100x550.jpg",
            mimeType: "image/jpeg"
          },
          {
            width: "550",
            file: "create-candy-cane-lettering-in-procreate-hero-550x1080.jpg",
            height: "1080",
            name: "readanddigest_large_height",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-550x1080.jpg",
            mimeType: "image/jpeg"
          },
          {
            width: "421",
            file: "create-candy-cane-lettering-in-procreate-hero-421x203.jpg",
            height: "203",
            name: "wp_rp_thumbnail",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-421x203.jpg",
            mimeType: "image/jpeg"
          }
        ]
      },
      mimeType: "image/jpeg",
      altText: "Create Candy Cane Lettering in Procreate",
      sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero.jpg",
      srcSet: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-500x281.jpg 500w, https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-1440x810.jpg 1440w, https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-1024x576.jpg 1024w, https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-1536x864.jpg 1536w, https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-100x56.jpg 100w, https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-1200x675.jpg 1200w, https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero.jpg 1920w",
      sizes: "(max-width: 500px) 100vw, 500px",
      id: "cG9zdDoxMDA4NQ=="
  }