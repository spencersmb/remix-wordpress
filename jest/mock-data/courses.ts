import { mockFeatureImageComplete } from "./images";

export const mockCourseRaw:ICourseRaw = {
  id: 1,
  details:{
    courseTags: [
      {
        tag: 'tag1'
      }
    ],
    courseUrl: 'https://www.example.com'
  },
  slug: 'course-slug',
  title: 'Course Title',
  featuredImage: {
    node: {
      ...mockFeatureImageComplete
    }
  }
}

export const mockCourse: ICourse = {
  id: 1,
  details:{
    courseTags: ['tag1'],
    courseUrl: 'https://www.example.com'
  },
  slug: 'course-slug',
  title: 'Course Title',
  featuredImage: mockFeatureImageComplete
}

export const mockCourse2: ICourse = {
  id: 2,
  details:{
    courseTags: ['tag2'],
    courseUrl: 'https://www.example2.com'
  },
  slug: 'course2-slug',
  title: 'Course2 Title',
  featuredImage: mockFeatureImageComplete
}

export const mockCourse3: ICourse = {
  id: 3,
  details:{
    courseTags: ['tag3'],
    courseUrl: 'https://www.example3.com'
  },
  slug: 'course3-slug',
  title: 'Course3 Title',
  featuredImage: mockFeatureImageComplete
}