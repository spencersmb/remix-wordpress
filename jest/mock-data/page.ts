import { mockFeaturedImage } from "./images";
import { mockAuthorData, mockSeo } from "./posts";

export const mockPage: IPage = {
  seo: {
    title: "Page Title",
    metaDesc: "Page description",
    opengraphModifiedTime: "2020-01-01T00:00:00.000Z",
    opengraphPublishedTime: "2020-01-01T00:00:00.000Z",
    readingTime: 1
  },
  title: "Page Title",
  slug: "page-slug",
  content: "Page Content",
  author: mockAuthorData,
  featuredImage: mockFeaturedImage,
  date: "2020-01-01",
  id: "page-id",
}