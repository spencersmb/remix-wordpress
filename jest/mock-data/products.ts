import { LicenseEnum } from "@App/enums/products";
import { mockFeaturedImage } from "./images";
import { mockSeo } from "./posts";

export const mockStandardLicense:ILicense = {
  licenseType: LicenseEnum.STANDARD,
  price: 15,
  url: "https://gum.co/beautiful-lettering"
}
export const mockExtendedLicense:ILicense = {
  licenseType: LicenseEnum.EXTENDED,
  price: 30,
  url: "https://gum.co/beautiful-lettering-extended"
}
export const mockServerLicense:ILicense = {
  licenseType: LicenseEnum.SERVER,
  price: 60,
  url: "https://gum.co/beautiful-lettering-server"
}
export const mockGenericProduct: IProduct = {
    title: 'product 1',
    slug: 'product-1',
    featuredImage: {
      node: mockFeaturedImage
    },
    productDetails: {
      font: {
        name: ''
      },
      licences: [

      ],
      productContent: {
        description: '',
        productfeatureimage: mockFeaturedImage,
        subtitle: ''
      },
      title: 'product 1',
      youtube: {
        url: ''
      }
    },
    seo: {
      metaDesc: '',
      opengraphModifiedTime: '',
      opengraphPublishedTime: '',
      readingTime: 1,
      title: '',
      opengraphImage: {
        altText: '',
        sourceUrl: '',
        id: '',
      },
      schema: {
        raw: ''
      },
      fullHead: ''
    }
}

export const mockPaidProduct: IProduct = {
  title: "Beautiful Lettering Brush Set",
  slug: "beautiful-lettering-brush-set",
  seo:{
    title: "Beautiful Lettering Brush Set",
    metaDesc: "Beautiful Lettering Brush Set",
    opengraphModifiedTime: "2020-01-01T00:00:00.000Z",
    opengraphPublishedTime: "2020-01-01T00:00:00.000Z",
    readingTime: 1,
    schema: {
      raw: 'string'
    },
    opengraphImage: {
      id: 'string',
      altText: 'string',
      sourceUrl: 'string'
    }
  },
  featuredImage: {
    node: {
      mimeType: "image/jpeg",
      mediaDetails: {
        height: 335,
        width: 435,
        sizes: [
          {
            width: "20",
            file: "wp-gumroad-brush-set-20x20.jpg",
            height: "20",
            name: "placeholder",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/wp-gumroad-brush-set-20x20.jpg",
            mimeType: "image/jpeg"
          },
          {
            width: "150",
            file: "wp-gumroad-brush-set-150x150.jpg",
            height: "150",
            name: "thumbnail",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/wp-gumroad-brush-set-150x150.jpg",
            mimeType: "image/jpeg"
          },
          {
            width: "100",
            file: "wp-gumroad-brush-set-100x77.jpg",
            height: "77",
            name: "et_thumbnail",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/wp-gumroad-brush-set-100x77.jpg",
            mimeType: "image/jpeg"
          },
          {
            width: "128",
            file: "wp-gumroad-brush-set-128x86.jpg",
            height: "86",
            name: "readanddigest_thumb",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/wp-gumroad-brush-set-128x86.jpg",
            mimeType: "image/jpeg"
          },
          {
            width: "421",
            file: "wp-gumroad-brush-set-421x203.jpg",
            height: "203",
            name: "wp_rp_thumbnail",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/wp-gumroad-brush-set-421x203.jpg",
            mimeType: "image/jpeg"
          }
        ]
      },
      altText: "Beautiful Lettering Brush Set",
      sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/wp-gumroad-brush-set.jpg",
      srcSet: "https://etheadless.local/wp-content/uploads/2021/11/wp-gumroad-brush-set.jpg 435w, https://etheadless.local/wp-content/uploads/2021/11/wp-gumroad-brush-set-100x77.jpg 100w",
      sizes: "(max-width: 435px) 100vw, 435px",
      id: "cG9zdDoxMDA1Mg=="
    }
  },
  productDetails: {
    font:{
      name: "Arial",
    },
    title: "Beautiful Lettering Brush Set",
    productContent:{
      description: null,
      productfeatureimage: {
        ...mockFeaturedImage
      },
      subtitle:null,
    },
    youtube:{
      url: ''
    },
    licences: [
      {
        licenseType: LicenseEnum.EXTENDED,
        price: 30,
        url: "https://gum.co/beautiful-lettering-extended"
      },
      {
        licenseType: LicenseEnum.STANDARD,
        price: 15,
        url: "https://gum.co/beautiful-lettering"
      }
    ]
  }
}