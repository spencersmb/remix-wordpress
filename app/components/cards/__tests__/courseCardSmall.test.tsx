import { mockCourse } from '@TestUtils/mock-data/courses';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import CourseCardSmall from '../courseCardSmall';

it('Should render Small Course Card Component ', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <CourseCardSmall course={mockCourse} />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
})