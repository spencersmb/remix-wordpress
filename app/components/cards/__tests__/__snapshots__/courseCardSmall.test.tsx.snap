// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Small Course Card Component  1`] = `
<a
  className="relative flex flex-row items-center p-4 mb-4 transition-all duration-300 translate-y-0 z-3 bg-tangerine-200 group laptop:hover:shadow-et_4 laptop:hover:z-2 laptop:hover:-translate-y-1"
  href="https://www.example.com"
  rel="noreferrer"
  target="_blank"
>
  <div
    className="rounded-full w-[113px] h-[113px] overflow-hidden border-0 transition-all duration-200"
  >
    <div
      className="relative flex-1"
      data-testid="padding-bot"
      style={
        Object {
          "paddingBottom": "100%",
        }
      }
    >
      <div
        className="absolute w-full lazy-load-wrapper lazy-load-wrapper-block lazy-load-image-full"
      >
        <span
          className=" lazy-load-image-background blur"
          style={
            Object {
              "backgroundImage": "url(https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-20x20.jpg)",
              "backgroundSize": "100% 100%",
              "color": "transparent",
              "display": "inline-block",
              "height": "150px",
              "width": "150px",
            }
          }
        >
          <span
            className=""
            style={
              Object {
                "display": "inline-block",
                "height": "150px",
                "width": "150px",
              }
            }
          />
        </span>
      </div>
    </div>
  </div>
  <div
    className="flex flex-col flex-1 ml-4"
  >
    <div
      className="text-xs font-semibold uppercase text-emerald-900"
    >
      course
    </div>
    <div
      className="text-xl font-semibold leading-7 text-emerald-900"
    >
      Course Title
    </div>
  </div>
</a>
`;
